src/commands/templates/delete.ts

```diff
@@ -0,0 +1,225 @@
+
+interface TemplateDeleteArguments {
+  serverToken: string;
+  requestHost: string;
+}
+
+export interface TemplateListOptions {
+  requestHost: string;
+  sourceServer: string;
+  templateType: TemplateTypes | undefined;
+}
+
+export async function handler(args: TemplateDeleteArguments): Promise<void> {
+  const serverToken = await validateToken(args.serverToken);
+  const requestHost = args.requestHost;
+  deletePrompts(serverToken, requestHost);
+}
+
+/**
+ * Ask user a series of questions before deleting templates.
+ */
+
+async function deletePrompts(
+  serverToken: string,
+  requestHost: string
+): Promise<void> {
+  const choice = await select({
+    message: "Choose how you want to delete templates:",
+    choices: [
+      {
+        name: "Delete templates by id or alias",
+        description:
+          "NOTE: mixing ids and aliases on the same request is NOT supported",
+        value: "byIdOrAlias",
+      },
+      {
+        name: "Delete all templates",
+        value: "all",
+      },
+    ],
+  });
+
+  /**
+   * if the user has selected to delete templates by id
+   */
+
+  if (choice === "byIdOrAlias") {
+    // TODO we probably want an actual validation
+    // against ids/aliases that exist in the server
+    const validateUserInput = async (input: string) => {
+      // do not allow empty strings
+      if (input) return true;
+      return false;
+    };
+
+    const userInput = await input({
+      message:
+        "Enter template id(s)/alias(es) - separate with commas if multiple:",
+      validate: validateUserInput,
+    });
+
+    const templateIdsOrAliases = userInput.split(",");
+
+    // if user has entered at least one id
+    if (templateIdsOrAliases.length) {
+      return deleteTemplates(templateIdsOrAliases, {
+        requestHost: requestHost,
+        sourceServer: serverToken,
+        templateType: undefined,
+      });
+    }
+  }
+
+  /**
+   * if the user has selected to delete all templates
+   */
+
+  if (choice === "all") {
+    let templateType, confirmed, inputIsValid;
+
+    templateType = await select({
+      message: "Which template type do you want to delete?",
+      choices: [
+        {
+          name: "Templates",
+          value: TemplateTypes.Standard,
+        },
+        {
+          description:
+            "NOTE: Layouts can only be deleted if not in use by any Templates",
+          name: "Layouts",
+          value: TemplateTypes.Layout,
+        },
+      ],
+    });
+
+    if (templateType) {
+      confirmed = await confirm({
+        default: false,
+        message: `Delete ALL templates? Are you sure?`,
+      });
+    }
+
+    // user has to type the exact sentence to proceed
+    const validateInput = async (input: string) =>
+      input !== "delete all templates" ? false : true;
+
+    if (confirmed) {
+      inputIsValid = await input({
+        message: 'Enter "delete all templates" to confirm:',
+        validate: validateInput,
+      });
+    }
```

Comment:
we could inline the validation here too:

```suggestion
    if (confirmed) {
      inputIsValid = await input({
        message: 'Enter "delete all templates" to confirm:',
        validate: (s) => s === 'delete all templates',
      });
    }
```